Generating updateByXInList broken when updating from micronaut 2.5.3 to 3.2.5

### Expected Behavior

When using updateByIdInList as a method name we expect the method to update every record that has its id in the provided list of ids be updated with the remaining provided parameters (which worked with micronaut 2.5.3).

The generated query looked like this:

```
13:16:35.320 [Test worker] DEBUG io.micronaut.data.query - Executing Query: UPDATE `Book` SET `topic`=? WHERE (`id` IN (?,?,?))
```

### Actual Behaviour

At least since version 3.2.5 of micronaut we get errors with this method. 

```
Unable to prepare query [UPDATE `Book` SET `id`=?,?,?,`topic`=? WHERE (`id` IN (?,?,?))]: Syntax Fehler in SQL Befehl "UPDATE `BOOK` SET `ID`=?,?[*],?,`TOPIC`=? WHERE (`ID` IN (?,?,?))"; erwartet "identifier"
Syntax error in SQL statement "UPDATE `BOOK` SET `ID`=?,?[*],?,`TOPIC`=? WHERE (`ID` IN (?,?,?))"; expected "identifier"; SQL statement:
UPDATE `Book` SET `id`=?,?,?,`topic`=? WHERE (`id` IN (?,?,?)) [42001-200]
io.micronaut.data.exceptions.DataAccessException: Unable to prepare query [UPDATE `Book` SET `id`=?,?,?,`topic`=? WHERE (`id` IN (?,?,?))]: Syntax Fehler in SQL Befehl "UPDATE `BOOK` SET `ID`=?,?[*],?,`TOPIC`=? WHERE (`ID` IN (?,?,?))"; erwartet "identifier"
Syntax error in SQL statement "UPDATE `BOOK` SET `ID`=?,?[*],?,`TOPIC`=? WHERE (`ID` IN (?,?,?))"; expected "identifier"; SQL statement:
UPDATE `Book` SET `id`=?,?,?,`topic`=? WHERE (`id` IN (?,?,?)) [42001-200]
	at io.micronaut.data.runtime.operations.internal.AbstractSqlRepositoryOperations.prepareStatement(AbstractSqlRepositoryOperations.java:166)
	at io.micronaut.data.jdbc.operations.DefaultJdbcRepositoryOperations.lambda$executeUpdate$8(DefaultJdbcRepositoryOperations.java:646)
	at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.executeWrite(AbstractSynchronousTransactionManager.java:175)
	at io.micronaut.data.jdbc.operations.DefaultJdbcRepositoryOperations.executeUpdate(DefaultJdbcRepositoryOperations.java:643)
	at io.micronaut.data.runtime.intercept.DefaultUpdateInterceptor.intercept(DefaultUpdateInterceptor.java:46)
	at io.micronaut.data.intercept.DataIntroductionAdvice.intercept(DataIntroductionAdvice.java:114)
	at io.micronaut.data.intercept.DataIntroductionAdvice.intercept(DataIntroductionAdvice.java:88)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137)
	at io.micronaut.validation.ValidatingInterceptor.intercept(ValidatingInterceptor.java:138)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137)
	at de.ksmwsk.deleteall.BookRepository$Intercepted.updateByIdInList(Unknown Source)
	at de.ksmwsk.deleteall.GetBooksIT.readsBooks(GetBooksIT.java:35)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.extension.InvocationInterceptor.interceptTestMethod(InvocationInterceptor.java:117)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:102)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:82)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:78)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:132)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:175)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:157)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax Fehler in SQL Befehl "UPDATE `BOOK` SET `ID`=?,?[*],?,`TOPIC`=? WHERE (`ID` IN (?,?,?))"; erwartet "identifier"
Syntax error in SQL statement "UPDATE `BOOK` SET `ID`=?,?[*],?,`TOPIC`=? WHERE (`ID` IN (?,?,?))"; expected "identifier"; SQL statement:
UPDATE `Book` SET `id`=?,?,?,`topic`=? WHERE (`id` IN (?,?,?)) [42001-200]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:453)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:429)
	at org.h2.message.DbException.getSyntaxError(DbException.java:243)
	at org.h2.command.Parser.readColumnIdentifier(Parser.java:4976)
	at org.h2.command.Parser.readTableColumn(Parser.java:1236)
	at org.h2.command.Parser.parseUpdateSetClause(Parser.java:1325)
	at org.h2.command.Parser.parseUpdate(Parser.java:1290)
	at org.h2.command.Parser.parsePrepared(Parser.java:1005)
	at org.h2.command.Parser.parse(Parser.java:843)
	at org.h2.command.Parser.parse(Parser.java:815)
	at org.h2.command.Parser.prepareCommand(Parser.java:738)
	at org.h2.engine.Session.prepareLocal(Session.java:657)
	at org.h2.engine.Session.prepareCommand(Session.java:595)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1235)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:352)
	at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:337)
	at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java)
	at io.micronaut.data.runtime.operations.internal.AbstractSqlRepositoryOperations.prepareStatement(AbstractSqlRepositoryOperations.java:164)
	... 103 more
```

It seems like the provided list is also used in the SET portion of the generated query now.

### Steps To Reproduce

- Create a simple application with a simple repository
- Try to use a method of the form updateByFieldNameInList to batch update multiple entries at once with the same value

### Environment Information

OS: Does not apply
Micronaut version: 2.5.3 vs. 3.2.5


### Example Application

https://github.com/ckosmowski/micronaut-examples/tree/issues/updateinlist

### Version

3.2.5

--- a/data-model/src/main/java/io/micronaut/data/model/jpa/criteria/impl/CriteriaUtils.java
+++ b/data-model/src/main/java/io/micronaut/data/model/jpa/criteria/impl/CriteriaUtils.java
@@ -25,6 +25,7 @@ import io.micronaut.data.model.jpa.criteria.PersistentPropertyPath;
 import io.micronaut.data.model.jpa.criteria.impl.predicate.ConjunctionPredicate;
 import io.micronaut.data.model.jpa.criteria.impl.predicate.DisjunctionPredicate;
 import io.micronaut.data.model.jpa.criteria.impl.predicate.PersistentPropertyBinaryPredicate;
+import io.micronaut.data.model.jpa.criteria.impl.predicate.PersistentPropertyInValuesPredicate;
 import jakarta.persistence.criteria.Expression;
 import jakarta.persistence.criteria.ParameterExpression;
 
@@ -180,6 +181,13 @@ public final class CriteriaUtils {
             PersistentPropertyBinaryPredicate<?> pp = (PersistentPropertyBinaryPredicate<?>) predicate;
             if (pp.getExpression() instanceof ParameterExpression) {
                 parameters.add((ParameterExpression<?>) pp.getExpression());
+            }
+        } else if (predicate instanceof PersistentPropertyInValuesPredicate) {
+            PersistentPropertyInValuesPredicate<?> pp = (PersistentPropertyInValuesPredicate<?>) predicate;
+            for (Expression<?> expression : pp.getValues()) {
+                if (expression instanceof ParameterExpression) {
+                    parameters.add((ParameterExpression<?>) expression);
+                }
             }
         } else if (predicate instanceof ConjunctionPredicate) {
             ConjunctionPredicate conjunctionPredicate = (ConjunctionPredicate) predicate;

--- a/data-processor/src/test/groovy/io/micronaut/data/processor/sql/BuildUpdateSpec.groovy
+++ b/data-processor/src/test/groovy/io/micronaut/data/processor/sql/BuildUpdateSpec.groovy
@@ -68,7 +68,7 @@ interface PersonRepository extends CrudRepository<Person, Long> {
     @Unroll
     void "test build update with datasource set"() {
         given:
-        def repository = buildRepository('test.MovieRepository', """
+            def repository = buildRepository('test.MovieRepository', """
 import io.micronaut.data.jdbc.annotation.JdbcRepository;
 import io.micronaut.data.model.query.builder.sql.Dialect;
 
@@ -76,6 +76,7 @@ import io.micronaut.data.model.query.builder.sql.Dialect;
 @io.micronaut.context.annotation.Executable
 interface MovieRepository extends CrudRepository<Movie, Integer> {
     void updateById(int id, String theLongName, String title);
+    void updateByIdInList(List<Integer> id, String title);
     void updateAll(java.util.List<Movie> movies);
 }
 
@@ -89,10 +90,11 @@ ${entity('Movie', [title: String, theLongName: String])}
         getParameterBindingIndexes(method) == binding
 
         where:
-        methodName   | query                                                             | bindingPaths                               | binding
-        'update'     | 'UPDATE `movie` SET `title`=?,`the_long_name`=? WHERE (`id` = ?)' | ['title', 'theLongName', 'id'] as String[] | ['-1', '-1', '-1'] as String[]
-        'updateById' | 'UPDATE `movie` SET `the_long_name`=?,`title`=? WHERE (`id` = ?)' | ['theLongName', 'title', 'id'] as String[] | ['1', '2', '0'] as String[]
-        'updateAll'  | 'UPDATE `movie` SET `title`=?,`the_long_name`=? WHERE (`id` = ?)' | ['title', 'theLongName', 'id'] as String[] | ['-1', '-1', '-1'] as String[]
+        methodName         | query                                                             | bindingPaths                               | binding
+        'update'           | 'UPDATE `movie` SET `title`=?,`the_long_name`=? WHERE (`id` = ?)' | ['title', 'theLongName', 'id'] as String[] | ['-1', '-1', '-1'] as String[]
+        'updateById'       | 'UPDATE `movie` SET `the_long_name`=?,`title`=? WHERE (`id` = ?)' | ['theLongName', 'title', 'id'] as String[] | ['1', '2', '0'] as String[]
+        'updateAll'        | 'UPDATE `movie` SET `title`=?,`the_long_name`=? WHERE (`id` = ?)' | ['title', 'theLongName', 'id'] as String[] | ['-1', '-1', '-1'] as String[]
+        'updateByIdInList' | 'UPDATE `movie` SET `title`=? WHERE (`id` IN (?))'                | ['title','id'] as String[]                 | ['1', '0'] as String[]
     }
 
     @Unroll

--- a/data-tck/src/main/groovy/io/micronaut/data/tck/tests/AbstractRepositorySpec.groovy
+++ b/data-tck/src/main/groovy/io/micronaut/data/tck/tests/AbstractRepositorySpec.groovy
@@ -431,6 +431,24 @@ abstract class AbstractRepositorySpec extends Specification {
         peopleUpdated.get(1).name.endsWith(" X")
     }
 
+    void "test update many2"() {
+        given:
+        saveSampleBooks()
+
+        when:
+        def books = bookRepository.findByAuthorName("Stephen King")
+        then:
+        books.size() == 2
+
+        when:
+        bookRepository.updateByIdInList(books.collect { it.id }, "Modified")
+        books = bookRepository.findByAuthorName("Stephen King")
+
+        then:
+        books[0].title == "Modified"
+        books[1].title == "Modified"
+    }
+
     void "test custom insert"() {
         given:
         personRepository.saveCustom([new Person(name: "Abc", age: 12), new Person(name: "Xyz", age: 22)])

--- a/data-tck/src/main/java/io/micronaut/data/tck/repositories/BookRepository.java
+++ b/data-tck/src/main/java/io/micronaut/data/tck/repositories/BookRepository.java
@@ -107,6 +107,8 @@ public abstract class BookRepository implements PageableRepository<Book, Long> {
 
     public abstract long updateAuthor(@Parameter("id") @Id Long id, @Parameter("author") Author author);
 
+    public abstract void updateByIdInList(List<Long> id, String title);
+
     public void saveAuthorBooks(List<AuthorBooksDto> authorBooksDtos) {
         List<Author> authors = new ArrayList<>();
         for (AuthorBooksDto dto: authorBooksDtos) {

